src/main/cpp/SwerveDrive.cpp

```diff
@@ -1,122 +1,122 @@
     }
 
-    double a = x1 - x2 * (m_L/r);
-    double b = x1 + x2 * (m_L/r);
-    double c = y1 - x2 * (m_W/r);
-    double d = y1 + x2 * (m_W/r);
+    double a = x1 - x2 * (m_L / r);
+    double b = x1 + x2 * (m_L / r);
+    double c = y1 - x2 * (m_W / r);
+    double d = y1 + x2 * (m_W / r);
 
-    double backRightSpeed = sqrt((a*a) + (d*d));
-    double backLeftSpeed = sqrt((a*a) + (c*c));
-    double frontRightSpeed = sqrt((b*b) + (d*d));
-    double frontLeftSpeed = sqrt((b*b) + (c*c));
+    double backRightSpeed = sqrt((a * a) + (d * d));
+    double backLeftSpeed = sqrt((a * a) + (c * c));
+    double frontRightSpeed = sqrt((b * b) + (d * d));
+    double frontLeftSpeed = sqrt((b * b) + (c * c));
 
-    double backRightAngle = atan2(a, d) * 180/ M_PI;
-    double backLeftAngle = atan2(a, c) *180/ M_PI;
-    double frontRightAngle = atan2(b, d) *180/ M_PI;
-    double frontLeftAngle = atan2(b, c) *180/ M_PI;
+    double backRightAngle = atan2(a, d) * 180 / M_PI;
+    double backLeftAngle = atan2(a, c) * 180 / M_PI;
+    double frontRightAngle = atan2(b, d) * 180 / M_PI;
+    double frontLeftAngle = atan2(b, c) * 180 / M_PI;
 
-    if(x2 == 0 && x1 == 0 && y1 == 0){
+    if (x2 == 0 && x1 == 0 && y1 == 0) {
         m_backRight.Stop();
         m_backLeft.Stop();
         m_frontRight.Stop();
         m_frontLeft.Stop();
         return;
     }
 
-    m_backRight.drive(-backRightSpeed, backRightAngle,
-        DriveConstants::BROFF);
-    m_backLeft.drive(backLeftSpeed, backLeftAngle,
-        DriveConstants::BLOFF);
+    m_backRight.drive(-backRightSpeed, backRightAngle, DriveConstants::BROFF);
+    m_backLeft.drive(backLeftSpeed, backLeftAngle, DriveConstants::BLOFF);
     m_frontRight.drive(-frontRightSpeed, frontRightAngle,
-        DriveConstants::FROFF);
-    m_frontLeft.drive(frontLeftSpeed, frontLeftAngle, 
-        DriveConstants::FLOFF);
-}
-
-
-//Updates the Odometry
-void
-SwerveDrive::UpdateOdometry(double theta){
-    double BR_A = m_backRight.getAngle(DriveConstants::BROFF);
-    double BR_V = m_backRight.getVelocity();
-    double BL_A = m_backLeft.getAngle(DriveConstants::BLOFF);
-    double BL_V = m_backLeft.getVelocity();
-    double FR_A = m_frontRight.getAngle(DriveConstants::FROFF);
-    double FR_V = m_frontRight.getVelocity();
-    double FL_A = m_frontLeft.getAngle(DriveConstants::FLOFF);
-    double FL_V = m_frontLeft.getVelocity();
-    m_odometry.updateOdometry(BR_A, BR_V, BL_A, BL_V, FR_A, FR_V,
-        FL_A, FL_V, theta);
-}
-
-
-//Trajectory Following for Swerve
-//Cannot turn and move at the same time
-//If you want to turn make Waypoint as 0,0,0, theta
-void
-SwerveDrive::TrajectoryFollow(double rot, size_t waypointIndex){
+                       DriveConstants::FROFF);
+    m_frontLeft.drive(frontLeftSpeed, frontLeftAngle, DriveConstants::FLOFF);
+}
+
+// Updates the Odometry
+void SwerveDrive::UpdateOdometry(double theta_rad, double dt) {
+    // NOTE I am explicitly ignoring any polarity flips,

```

Comment:
This is the only substantial change to the swerve, make close note of this comment btw